### ISSUE TITLE
upgrade from surface_reflectance_june2017 updatelads.py

### ISSUE THREAD

**Issue by mquinteiro** [USGS-EROS/espa-surface-reflectance#93]
I'm trying to upgrade to a new version because I can't update LaSRC auxiliary files anymore (ftp download changes)
I have update to the new version surface_reflectance_may2018 but I'm absolutely lost 

updatelads.py ask for a file called ~/.usgs/espa/processing.conf and I don't know where it come front or how to make it. Where is it?

I didn't remake all the project, unfortunately I didn't save all environment variables and I'm scared about rebuild all the project. It is needed?

**Comment by mquinteiro** [comment]
Editing the updatelads.py and adding the token in the var TOKEN is enough to start to work... But....

The script fails downloading MYD09CMG producto, because malformed URL
example;
```
2018-05-30 12:33:59.205 2441 INFO     updatelads.py:114:geturl -- Retry 4 of download for https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/6/MYD09CMG/2018/138//MYD09CMG.A2018138.006.2018140025537.hdf
2018-05-30 12:34:50.130 2441 WARNING  updatelads.py:440:getLadsData -- No Aqua or Terra LAADS CMA data available for doy 138 year 2018. Skipping this date.

```
The double / looks faulty, if I manually try https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/6/MYD09CMG/2018/138/MYD09CMG.A2018138.006.2018140025537.hdf works perfect

**Comment by gschmidt-usgs** [comment]
Looks like you read the information that specifies you need a TOKEN to download the LAADS data in the non-ESPA environment.  The double / works fine.  Yes, it could be changed, but it does work ok.  In working with this https URL over the past week, we have found that the https service is up and down at times.

**Comment by mquinteiro** [comment]
@gschmidt-usgs I get it work with TOKEN, but still having problems, 
```
2018-05-30 13:25:45.610 18410 INFO     updatelads.py:198:downloadLads -- Cleaning download directory: /tmp/lads/2018
2018-05-30 13:25:45.710 18410 INFO     updatelads.py:215:downloadLads -- Downloading data for 2018/141 to /tmp/lads/2018
2018-05-30 13:25:45.710 18410 INFO     updatelads.py:218:downloadLads -- Retrieving https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/6/MOD09CMA2018/141/ to /tmp/lads/2018
curl: (22) The requested URL returned error: 400 Bad Request
2018-05-30 13:25:47.060 18410 ERROR    updatelads.py:125:geturl -- curl GET error for URL (). https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/6/MOD09CMA2018/141/.csv:
```

Take a look at this "a/6/MOD09CMA2018/141/.csv:" the slash is not needed, that makes the first error. Still thinking that there is a / more
         import csv
            files = [f for f in csv.DictReader(StringIO(geturl('%s.csv' % url, token)), skipinitialspace=True)]
and url comes from buidURLs which have:

  ```
  # append TERRA CMA data (MOD09CMA)
    url = ('{}{}{}/{:03d}/'.format(SERVER_URL, TERRA_CMA, year, doy))
    urlList.append(url)

    # append TERRA CMG data (MOD09CMG)
    url = ('{}{}{}/{:03d}/'.format(SERVER_URL, TERRA_CMG, year, doy))
    urlList.append(url)

    # append AQUA CMA data (MYD09CMA)
    url = ('{}{}{}/{:03d}/'.format(SERVER_URL, AQUA_CMA, year, doy))
    urlList.append(url)

    # append AQUA CMG data (MYD09CMG)
    url = ('{}{}{}/{:03d}/'.format(SERVER_URL, AQUA_CMG, year, doy))
    urlList.append(url)

    return urlList


```

That means that is impossible to build the correct doy.csv

**Comment by gschmidt-usgs** [comment]
Just to confirm, you have replaced 'None' for TOKEN with the TOKEN you received from LAADS DAAC in the updatelads.py script?

##NOTE: For non-ESPA environments, the TOKEN needs to be defined.  This is
##the application token that is required for accessing the LAADS data
##https://ladsweb.modaps.eosdis.nasa.gov/tools-and-services/data-download-scripts/
TOKEN = None
USERAGENT = 'espa.cr.usgs.gov/updatelads.py 1.4.1--' + sys.version.replace('\n','').replace('\r','')

**Comment by mquinteiro** [comment]
Yes sorry I thought that this was clear since my 2nd message, I found the variable inside the code and replace none by one api-key created from my profile of ladsweb. Without that I can't advance. I have review the code and it can't work without the modification that I propose.
The modification that I did works, I can't understand who can it work for you. Do you want a pull request?

**Comment by gschmidt-usgs** [comment]
Sure, send a pull request and we will take a look at it.  Thanks.
